src/tests/test_pubkey_pqc.h

```diff
@@ -77,47 +90,54 @@ class PK_PQC_KEM_KAT_Test : public PK_Test {
    #endif
       }
 
-      Test::Result run_one_test(const std::string& header, const VarMap& vars) final {
-         Test::Result result(Botan::fmt("PQC KAT for {} with parameters {}", algo_name(), header));
-         auto d = Delegate(header);
+      Test::Result run_one_test(const std::string& params, const VarMap& vars) final {
+         Test::Result result(Botan::fmt("PQC KAT for {} with parameters {}", algo_name(), params));
 
          // All PQC algorithms use this DRBG in their KAT tests to generate
          // their private keys. The amount of data that needs to be pulled from
          // the RNG for keygen and encapsulation is dependent on the algorithm
          // and the implementation.
          auto ctr_drbg = create_drbg(vars.get_req_bin("Seed"));
-         auto rng_keygen = d.rng_for_keygen(*ctr_drbg);
-         auto rng_encaps = d.rng_for_encapsulation(*ctr_drbg);
+         auto rng_keygen = rng_for_keygen(params, *ctr_drbg);
+         auto rng_encaps = rng_for_encapsulation(params, *ctr_drbg);
 
          // Key Generation
-         auto sk = Private_Key(rng_keygen, d.mode());
-         result.test_is_eq("Generated private key", d.map_value(sk.raw_private_key_bits()), vars.get_req_bin("SK"));
-         result.confirm("All prepared random bits used for key generation", rng_keygen.empty());
+         auto sk = Botan::create_private_key(algo_name(), rng_keygen, params);
+         result.test_is_eq("Generated private key",
+                           map_value(params, sk->raw_private_key_bits(), VarType::PrivateKey),
+                           vars.get_req_bin("SK"));
+         inspect_rng_after_keygen(params, rng_keygen, result);
 
          // Algorithm properties
-         result.test_eq("algorithm name", sk.algo_name(), algo_name());
```

Comment:
I think we should keep this. It checks that the class under test correctly identifies itself with the correct name. E.g. in Kyber this was broken for a while (as in: it identified as "Kyber512/768/1024" when in Botan the `algo_name()` doesn't contain the key length by convention).

Unfortunately, the algo name string is duplicated at several places across the code base. Therefore, it makes sense to pin it in the test.

Reply:
Sure. I thought the algo name was implicitly tested when creating the keys. But they are not used in these create calls, you are right 👍